### ISSUE TITLE
Fix issues with newer versions of bson and chrono

### ISSUE THREAD

**Issue by TrionProg** [mongodb-labs/mongo-rust-driver-prototype#214]
I have Cargo.toml
```
[dependencies]
serde="1.0.2"
serde_derive="*"
bincode="*"
redis = "*"
cdrs = "1.0.0-rc.1"
bson = "0.7.0"
mongodb = "0.2.7"
```

```
Compiling mongodb v0.2.8
error[E0308]: mismatched types
   --> /home/trionprog/.cargo/registry/src/github.com-1ecc6299db9ec823/mongodb-0.2.8/src/gridfs/file.rs:592:37
    |
592 |             file.upload_date = Some(*datetime);
    |                                     ^^^^^^^^^ expected struct `chrono::DateTime`, found struct `chrono::datetime::DateTime`
    |
    = note: expected type `chrono::DateTime<chrono::UTC>`
               found type `chrono::datetime::DateTime<chrono::offset::utc::UTC>`
    = help: here are some functions which might fulfill your needs:
            - .timezone()

error[E0277]: the trait bound `bson::Bson: std::convert::From<chrono::DateTime<chrono::UTC>>` is not satisfied
   --> /home/trionprog/.cargo/registry/src/github.com-1ecc6299db9ec823/mongodb-0.2.8/src/gridfs/file.rs:616:23
    |
616 |           let mut doc = doc! {
    |  _______________________^ starting here...
617 | |             "_id" => (self.id.clone()),
618 | |             "chunkSize" => (self.chunk_size),
619 | |             "length" => (self.len),
620 | |             "md5" => (self.md5.to_owned()),
621 | |             "uploadDate" => (self.upload_date.as_ref().unwrap().clone())
622 | |         };
    | |_________^ ...ending here: the trait `std::convert::From<chrono::DateTime<chrono::UTC>>` is not implemented for `bson::Bson`
    |
    = help: the following implementations were found:
              <bson::Bson as std::convert::From<f32>>
              <bson::Bson as std::convert::From<f64>>
              <bson::Bson as std::convert::From<&'a str>>
              <bson::Bson as std::convert::From<std::string::String>>
            and 16 others
    = note: required by `std::convert::From::from`
    = note: this error originates in a macro outside of the current crate

error[E0308]: mismatched types
   --> /home/trionprog/.cargo/registry/src/github.com-1ecc6299db9ec823/mongodb-0.2.8/src/topology/monitor.rs:125:38
    |
125 |             result.local_time = Some(*datetime);
    |                                      ^^^^^^^^^ expected struct `chrono::DateTime`, found struct `chrono::datetime::DateTime`
    |
    = note: expected type `chrono::DateTime<chrono::UTC>`
               found type `chrono::datetime::DateTime<chrono::offset::utc::UTC>`
    = help: here are some functions which might fulfill your needs:
            - .timezone()

```

I've edited my Cargo.toml
```
[dependencies]
serde="1.0.2"
serde_derive="*"
bincode="*"
redis = "*"
cdrs = "1.0.0-rc.1"
bson = ">=0.7.0,<0.7.1" #to use 0.7.0
mongodb = ">=0.2.7,<0.2.8" #to use 0.2.7
```

But then =(
```
error: failed to select a version for `bson` (required by `mongodb`):
all possible versions conflict with previously selected versions of `bson`
  version 0.7.0 in use by bson v0.7.0
  possible versions to select: 0.7.1
```

**Comment by Gedweb** [comment]
Confirm

**Comment by saghm** [comment]
Thanks for the report! I just updated the dependencies and pushed to master. Assuming it passes the CI, I'll push a new version to crates.io.

**Comment by saghm** [comment]
Just published 0.2.9. Thanks again for reporting the issue!
